server/data/data-api/src/main/java/org/apache/james/dlp/api/DLPConfigurationItem.java

```diff
@@ -0,0 +1,185 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.dlp.api;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+public class DLPConfigurationItem {
+
+    public static class Builder {
+        private static final boolean NOT_TARGETED = false;
+
+        private Optional<Boolean> targetsSender;
+        private Optional<Boolean> targetsRecipients;
+        private Optional<Boolean> targetsContent;
+        private Optional<String> explanation;
+        private Optional<String> expression;
+
+        public Builder() {
+            targetsSender = Optional.empty();
+            targetsRecipients = Optional.empty();
+            targetsContent = Optional.empty();
+            explanation = Optional.empty();
+            expression = Optional.empty();
+        }
+
+        public Builder targetsSender() {
+            this.targetsSender = Optional.of(true);
+            return this;
+        }
+
+        public Builder targetsRecipients() {
+            this.targetsRecipients = Optional.of(true);
+            return this;
+        }
+
+        public Builder targetsContent() {
+            this.targetsContent = Optional.of(true);
+            return this;
+        }
+
+        public Builder expression(String expression) {
+            this.expression = Optional.of(expression);
+            return this;
+        }
+
+        public Builder explanation(String explanation) {
+            this.explanation = Optional.of(explanation);
+            return this;
+        }
+
+        public DLPConfigurationItem build() {
+            Preconditions.checkState(expression.isPresent(), "`expression` is mandatory");
```

Comment:
Then why is it an Optional?

Reply:
Becasue it can be missing in the builder during the life cycle of the builder. I prefer an Optional to represent that the value is not yet supplied rather than `null`.